ice-adapter/src/main/java/com/faforever/iceadapter/gpgnet/GPGNetServer.java

```diff
@@ -220,10 +238,14 @@ private static void onGpgnetConnectionLost() {
     /**
      * Listens for incoming connections from a game instance
      */
-    private static void acceptThread() {
+    private void acceptThread() {
         while (!Thread.currentThread().isInterrupted()) {
             log.info("Listening for incoming connections from game");
-            try (Socket socket = serverSocket.accept()) {
+            try {
+                // The socket declaration must not be moved into a try-with-resources block, as the socket must not be
+                // closed. It is passed into the GPGNetClient.
+                Socket socket = serverSocket.accept();
```

Comment:
Maybe can create a constructor in the `GPGNetClient(ServerSocket serverSocket)` and get a socket in the constructor code. Assuming that the client should close the socket, and not try close in try/catch.

Reply:
I am not even sure why it is split up in 2 classes. 1 class is probably sufficient and then keeps the reference to the socket. But that is out of scope of this PR.